### DISCUSSION TITLE
Any way to generate swagger?

### DISCUSSION THREAD

**Discussion by Darhagonable** [cesanta/mongoose#2986] [Ask questions, get answers]
Is there any way to generate a swagger file during build time/in pipeline from the mongoose http endpoints and models?

**Comment by scaprile** [comment]
I'm sorry, what ?
Mongoose is C, you write the event handler, there are no http endpoints and models, there is an event handler that is called when an HTTP request is received.
What is Mongoose for you ?
Please see our [documentation](https://mongoose.ws/documentation/),  and follow the guidelines in our [tutorials](https://mongoose.ws/documentation/#tutorials-and-examples).

**Reply by Darhagonable** [reply]
Sorry for my ignorance!
I'm a frontend web developer and know our backend is written with mongoose C++.
In past projects, we’ve used C# ASP.NET along with tools like Swashbuckle to automatically generate Swagger documentation. So I was wondering if there was something similar to Swashbuckle for mongoose.

I now see though that I had misunderstood the role mongoose plays and its not strictly comparable to ASP.NET.

Thank you!

**Reply by scaprile** [reply]
Don't worry, I don't know a bit on that world...
Probably someone crafted a backend for you
